earth_enterprise/src/common/khTypes.h

```diff
@@ -19,6 +19,26 @@
 #define GEO_EARTH_ENTERPRISE_SRC_COMMON_KHTYPES_H_
 
 #include <assert.h>
+#include <mutex>
+#include <vector>
+
+template <typename T>
```

Comment:
Another pretty big shortcoming here is for loops over the container items can't really be protected without modification of the calling logic.